### ISSUE TITLE
error when setting sound driver

### ISSUE THREAD

**Issue by vatula** [flathub/com.valvesoftware.Steam.Utility.protontricks#8]
when setting sound driver to alsa with protontricks I get the error

> Executing load_sound alsa
Setting sound driver to alsa
Executing /home/atticdweller/.steam/steam/steamapps/common/Proton 4.11/dist/bin/wine regedit C:\windows\Temp\set-sound.reg
wineserver: using server-side synchronization.
0019:err:module:load_so_dll failed to load .so lib "/home/atticdweller/.local/share/Steam/steamapps/common/Proton 4.11/dist/bin/../lib64/wine/winebus.sys.so": libudev.so.0: cannot open shared object file: No such file or directory
0019:err:ntoskrnl:ZwLoadDriver failed to create driver L"\\Registry\\Machine\\System\\CurrentControlSet\\Services\\WineBus": c0000142
000d:fixme:service:scmdatabase_autostart_services Auto-start service L"WineBus" failed to start: 1114
Executing /home/atticdweller/.steam/steam/steamapps/common/Proton 4.11/dist/bin/wine64 regedit C:\windows\Temp\set-sound.reg
wineserver: using server-side synchronization.
0019:err:module:load_so_dll failed to load .so lib "/home/atticdweller/.local/share/Steam/steamapps/common/Proton 4.11/dist/bin/../lib64/wine/winebus.sys.so": libudev.so.0: cannot open shared object file: No such file or directory
0019:err:ntoskrnl:ZwLoadDriver failed to create driver L"\\Registry\\Machine\\System\\CurrentControlSet\\Services\\WineBus": c0000142
000d:fixme:service:scmdatabase_autostart_services Auto-start service L"WineBus" failed to start: 1114

I have vanilla Fedora 31 with Steam flatpak.
I have installed protontricks as per README.

**Comment by Matoking** [comment]
Does the command to set the audio driver succeed? The error message might not be fatal if everything else works.

I checked and libudev.so.0 doesn't exist inside the Steam Flatpak (you can test this by running `ldconfig -p | grep udev` both inside and outside the Flatpak), hence why winecfg fails to find it and prints the error.

**Comment by vatula** [comment]
I don't think it succeeds. See screenshots (it's a sequence). The second-to-last image shows the settings were not saved.

![image](https://user-images.githubusercontent.com/450851/73585843-92e13680-44f9-11ea-8cc5-b7daddce6b3f.png)
![image](https://user-images.githubusercontent.com/450851/73585851-a7bdca00-44f9-11ea-8564-dca9b69ef2a0.png)
![image](https://user-images.githubusercontent.com/450851/73585865-c2903e80-44f9-11ea-9a1f-562364969388.png)
![image](https://user-images.githubusercontent.com/450851/73585877-dd62b300-44f9-11ea-998b-a1e2ce629b06.png)
![image](https://user-images.githubusercontent.com/450851/73585889-f0758300-44f9-11ea-899d-4033512c155b.png)
![image](https://user-images.githubusercontent.com/450851/73585897-04b98000-44fa-11ea-96ab-975de2ccda0e.png)
![image](https://user-images.githubusercontent.com/450851/73585928-62e66300-44fa-11ea-8385-fbae53e091d7.png)

**Comment by vatula** [comment]
Also a question. If steam in it's **flatpak** does not include the library, then isn't it weird to require missing library for a **flatpak** protontricks? I realize it might be an upstream issue somewhere.

**Comment by Matoking** [comment]
All of the options in the menu you provided are unchecked by default; they are simply a list of actions you can run, and the listing doesn't check whether the actions have been executed already or not.

I also checked the Winetricks code, and all the `sound=alsa` option does is edit a single key in the Windows registry, which is something that doesn't require a working libudev library and the registry key is changed despite the warning. You can check it by running `protontricks <APPID> regedit` and checking the value for `HKEY_CURRENT_USER/Software/Wine/Drivers/Audio`.

The Wine binaries that are trying to load the libudev.so.0 file are shipped with Steam, so it makes sense to fix it there. The underlying Freedesktop SDK [ships a more feature limited version of the udev library](https://gitlab.com/freedesktop-sdk/freedesktop-sdk/issues/860) since the udev library [doesn't guarantee API stability](https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/merge_requests/1604#note_211411965), so any third-party fix would likely be fragile.

**Comment by gasinvein** [comment]
This looks like a Proton or steam runtime issue to me. If Proton is linked with `libudev.so.0`, then it should be shipped with Steam runtime. `libudev.so.0` is part of udev version <= 175 and modern distributions don't provide it.
